service/glacier/treehash.go

```diff
@@ -18,8 +18,8 @@ type Hash struct {
 //
 // See http://docs.aws.amazon.com/amazonglacier/latest/dev/checksum-calculations.html for more information.
 func ComputeHashes(r io.ReadSeeker) Hash {
-	r.Seek(0, 0)       // Read the whole stream
-	defer r.Seek(0, 0) // Rewind stream at end
+	start, _ := r.Seek(0, 1)  // Read the whole stream
```

Comment:
replacing the `1` with `io.SeekCurrent` may be good here

Reply:
:( io.SeekCurrent require's [go 1.7](https://golang.org/doc/go1.7#io).